mm2src/mm2_main/src/lp_network.rs

```diff
@@ -129,18 +129,64 @@ pub async fn p2p_event_process_loop(ctx: MmWeak, mut rx: AdexEventRx, i_am_relay
     }
 }
 
+/// Validate topic by checking it's prefix and coin, exclude if invalid.
+fn topic_prefix_and_coin_validation(ctx: &MmArc, mut topics: Vec<String>) -> Vec<String> {
+    if let Some(conf) = ctx.conf["coins"].as_array() {
+        topics.retain(|topic| {
+            let mut split = topic.split(TOPIC_SEPARATOR);
+
+            if split.clone().count() != 2 {
+                return false;
+            }
+
+            match split.next() {
+                Some(lp_swap::TX_HELPER_PREFIX) => {
+                    if let Some(coin) = split.next() {
+                        conf.iter().any(|c| c["coin"].as_str() == Some(coin))
+                    } else {
+                        false
+                    }
+                },
+                Some(lp_ordermatch::ORDERBOOK_PREFIX) => {
+                    if let Some(pair) = split.next() {
+                        match lp_ordermatch::parse_orderbook_pair_from_topic(pair) {
+                            Some((coin1, coin2)) => {
+                                conf.iter().any(|c| c["coin"].as_str() == Some(coin1))
```

Comment:
We can have a situation when other nodes (GUIs) have new coins in the config, which do not exist on other nodes yet. With this change, orderbook sync may stop working for these coins. Considering the comment above, `topic_prefix_and_coin_validation` seems not required at all. Also, these repeated iterations on each topic are quite an overhead. If this is required in the future, please consider creating a `HashSet` of coins tickers, that exist in config, upon MM2 initialization to have O(1) instead of O(N) for these checks.

Reply:
I guess then this function is not needed for now. Should I remove it ?

Reply:
Yes, please :slightly_smiling_face: